packages/api/src/controllers/animalController.js

```diff
@@ -0,0 +1,115 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Model, transaction } from 'objection';
+import AnimalModel from '../models/animalModel.js';
+import baseController from './baseController.js';
+import DefaultAnimalBreedModel from '../models/defaultAnimalBreedModel.js';
+import CustomAnimalBreedModel from '../models/customAnimalBreedModel.js';
+
+const animalController = {
+  getFarmAnimals() {
+    return async (req, res) => {
+      try {
+        const { farm_id } = req.headers;
+        const rows = await AnimalModel.query().where({ farm_id });
+        return res.status(200).send(rows);
+      } catch (error) {
+        console.error(error);
+        return res.status(500).json({
+          error,
+        });
+      }
+    };
+  },
+
+  addAnimals() {
```

Comment:
That's true, we might even want that for creating them from the UI so that not all work is lost if there's an error on one animal, although I imagine that'd be less likely to happen from the UI. We can definitely improve this logic when we work on the CSV upload, it's a great observation!